From 050cffbdc1f5cdb538e6800e2ee1c2df375d300d Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Fri, 29 May 2015 10:37:51 -0700
Subject: [PATCH 120/441] sdhci: Handle IRQ if HW Mutex is not available

If controller requires HW Mutex, but the NetIP subsystem is not yet
available, don't ignore the incoming IRQs.
---
 include/linux/mmc/sdhci.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/mmc/sdhci.h
+++ b/include/linux/mmc/sdhci.h
@@ -202,7 +202,7 @@ struct sdhci_host {
 void lock_emmc_access(void);
 void unlock_emmc_access(void);
 
-#define EMMC_HW_MUTEX_IS_LOCKED(host) (netss_driver_ready() &&\
+#define EMMC_HW_MUTEX_IS_LOCKED(host) (!netss_driver_ready() ||\
   hw_mutex_is_locked(HW_MUTEX_EMMC))
 
 #define LOCK_EMMC_HW_MUTEX(host) do{\
